pinot-common/src/main/java/com/linkedin/pinot/common/utils/request/RequestUtils.java

```diff
@@ -201,8 +207,24 @@ public static boolean isFitForStarTreeIndex(SegmentMetadata segmentMetadata, Fil
     // Ensure that none of the group-by columns are metric or skipped for materialization.
     GroupBy groupBy = brokerRequest.getGroupBy();
     if (groupBy != null) {
+      Set<String> allGroupByColumns = new HashSet<>();
```

Comment:
This could be refactored out as a utility function that returns all related columns to be checked.

Reply:
done, added tests for the function too